### PR TITLE
Support Adding an item to the cart

### PR DESCRIPTION
**Pull request by Deepak-Testpress**

### CODE REVIEW

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: myshop/tests/cart/test_cart.py [hunk 1]
```diff
@@ -90,7 +90,7 @@ def test_change_product_quantity_overwrites_existing_quantity_with_new_quantity(
 
         self.assertEqual(new_quantity, cart[self.product_id_str]["quantity"])
 
-    def test_change_product_quantity_adds_product_to_cart_with_given_quantity_when_product_not_already_exists_in_cart(
+    def test_change_product_quantity_adds_product_to_cart_with_quantity_when_product_not_already_exists_in_cart(
```
[the remark this answers is not in the document]
Comment (Deepak-Testpress) [reply myshop/tests/cart/test_cart.py]: Changed in the previous PR

File: myshop/cart/views.py [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+from shop.models import Product
+from .cart import Cart
+from .forms import CartAddProductForm
 
-# Create your views here.
+
+@require_POST
+def cart_add(request, product_id):
+    cart = Cart(request)
+    product = get_object_or_404(Product, id=product_id)
```
[the remark this answers is not in the document]
Comment (Deepak-Testpress) [reply myshop/cart/views.py:11]: Added test cases for cart_add method here

File: myshop/tests/cart/test_forms.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    ):
+        form = CartAddProductForm(
+            data={
+                "quantity": 24,
+            }
+        )
+
+        self.assertFalse(form.is_valid())
+
+    def test_cart_add_product_validates_false_with_quantity_undervalue_choice_range(
```
[the remark this answers is not in the document]
Comment (Deepak-Testpress) [reply myshop/tests/cart/test_forms.py]: Renamed

File: myshop/tests/cart/test_forms.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    ):
+        form = CartAddProductForm(
+            data={
+                "quantity": 20,
+            }
+        )
+
+        self.assertTrue(form.is_valid())
+
+    def test_cart_add_product_validates_false_with_quantity_exceeding_choice_range(
```
[the remark this answers is not in the document]
Comment (Deepak-Testpress) [reply myshop/tests/cart/test_forms.py]: Renamed

File: myshop/tests/cart/test_forms.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
+from django.test import TestCase
+from cart.forms import CartAddProductForm
+
+
+class TestForms(TestCase):
+    def test_cart_add_product_validates_true_with_quantity_not_exceeding_choice_range(
```
[the remark this answers is not in the document]
Comment (Deepak-Testpress) [reply myshop/tests/cart/test_forms.py]: Renamed

**Review by waranlogesh — APPROVED** [review]

**Pull request merged**